docs/transparent-validation/pikvm/assembly-and-validation.md

```diff
@@ -118,6 +118,148 @@ which is not equipped with an Ethernet port.
     might not connect to the WiFi. To prevent this, the router should be
     configured properly: channels 12-14 or Auto Scan mode should be disabled.
 
+### Alternative method of setting up WiFi
+
+The  section below describes how to configure wireless network connection if
+the method mentioned above does not bring expected results.
+
+1. Make the filesystem read - write by executing following command:
+
+    ```bash
+    rw
+    ```
+
+1. Run following command to check if wireless device is up:
+
+    ```bash
+    ip link show wlan0
+    ```
+
+    Example output:
+
+    ```bash
+    3: wlan0: (BROADCAST,MULTICAST) mtu 1500 qdisc noop state DOWN mode DEFAULT qlen 1000
+    link/ether 74:e5:43:a1:ce:65 brd ff:ff:ff:ff:ff:ff
+    ```
+
+    Look for the word "UP" inside the brackets in the first line of the output.
+
+1. In the above example, wlan0 is not UP. Execute the following command to bring
+    it up:
+
+    ```bash
+    ip link set wlan0 up
+    ```
+
+1. Scan to find out what WiFi network(s) are detected:
+
+    ```bash
+    /sbin/iw wlan0 scan
+    ```
+
+    The command will output all available wireless networks. Example output:
+
+    ```bash
+    BSS 60:38:e0:d6:46:99(on wlp0s20f3) -- associated
+	last seen: 1588.726s [boottime]
+	TSF: 2351559493374 usec (27d, 05:12:39)
+	freq: 2417
+	beacon interval: 100 TUs
+	capability: ESS Privacy ShortPreamble ShortSlotTime (0x0431)
+	signal: -77.00 dBm
+	last seen: 10532 ms ago
+	Information elements from Probe Response frame:
+	SSID: test_network
+	Supported rates: 1.0* 2.0* 5.5* 11.0* 22.0 6.0 9.0 12.0
+	DS Parameter set: channel 2
+	ERP: <no flags>
+	Extended supported rates: 18.0 24.0 36.0 48.0 54.0
+    ...
+    ```
+
+1. Generate a configuration file for wpa_supplicant that contains the
+    pre-shared key ("passphrase") for the WiFi network. Above network is going
+    to be used as an example. Run following command:
+
+    ```bash
+    wpa_passphrase test_network >> /etc/wpa_supplicant.conf
+    ```
+
+    After executing the command above, type in the password for the network.
+
+1. Check the wpa_supplicant configuration file using following command:
+
+    ```bash
+    cat /etc/wpa_supplicant/wpa_supplicant.conf
+    ```
+
+    Example output:
+
+    ```bash
+    # reading passphrase from stdin
+    network={
+	ssid="test_network"
+	#psk="testtest"
+	psk=4dfe1c985520d26a13e932bf0acb1d4580461dd854ed79ad1a88ec221a802061
+    }
+    ```
+
+1. Run wpa_supplicant with the new configuration file using following command:
+
+    ```bash
+    wpa_supplicant -B -D wext -i wlan0 -c /etc/wpa_supplicant/wpa_supplicant.conf
```

Comment:
@johanes2115 This procedure seems very manual. Also, I do not believe it works after PiKVM reboots, unless we enable service as in the official documentation: https://docs.pikvm.org/wifi/#step-by-step-advances-users-only-section